watch.go

```diff
@@ -176,6 +176,6 @@ func (e *Executor) registerWatchedFiles(w *watcher.Watcher, calls ...taskfile.Ca
 }
 
 func shouldIgnoreFile(path string) bool {
-	return strings.Contains(path, "/.git") || strings.Contains(path, "/.hg") ||
-		strings.Contains(path, "/.task") || strings.Contains(path, "/node_modules")
+	return strings.Contains(path, "/.git/") || strings.Contains(path, "/.hg/") ||
+		strings.Contains(path, "/.task/") || strings.Contains(path, "/node_modules/")
```

Comment:
This change will break in situations where the path ends in exactly `.git` (with no trailing slash). Can I suggest something like this: 

```suggestion
	ignorePaths := []string{
		"/.task",
		"/.git",
		"/.hg",
		"/.node_modules",
	}
	for _, p := range ignorePaths {
		if strings.Contains(path, fmt.Sprintf("%s/", p)) || strings.HasSuffix(path, p) {
			return true
		}
	}
	return false
```

Making this into a loop is also a bit cleaner than the current or statement IMO.

Reply:
Yeah, its actually solves everything!